Habitica/src/main/java/com/habitrpg/android/habitica/ui/activities/FullProfileActivity.kt

```diff
@@ -125,38 +121,32 @@ class FullProfileActivity : BaseActivity() {
                 bundleOf(Pair("userID", userID), Pair("username", null))
             )
         }
+
+        lifecycleScope.launch(ExceptionHandler.coroutine()) {
+            refresh()
+            invalidateOptionsMenu()
+        }
+    }
+
+    private suspend fun refresh() {
         lifecycleScope.launch(ExceptionHandler.coroutine()) {
             userRepository.getUser()
                 .collect {
                     blocks = it?.inbox?.blocks ?: listOf()
                     binding.blockedDisclaimerView.visibility =
                         if (isUserBlocked()) View.VISIBLE else View.GONE
-
                     isModerator = it?.hasPermission(Permission.MODERATOR) == true
                     binding.adminStatusView.isVisible = isModerator
-                    if (isModerator) {
-                        val member = socialRepository.retrieveMember(userID, true)
-                        member?.stats = this@FullProfileActivity.member.value?.stats
-                        if (member != null) {
-                            updateView(member)
-                        }
+
+                    //Get Current viewed profile user
+                    val member = socialRepository.retrieveMember(userID)

```

Comment:
We need to do two calls here for moderators. This returns the actual user data. But to display if a user is shadow muted or not, it also need to do the `socialRepository.retrieveMember(userID, true)` call to get the data from hall of heroes. Unfortunately that does not contain all info about the member. So we'd need to change it, so that the data returned from the hall of heroes call is only used to set the adminStatusView

Reply:
tyvm! That's super helpful - I'll make those changes